snuba/utils/streams/dummy.py

```diff
@@ -0,0 +1,106 @@
+from datetime import datetime
+from typing import (
+    Callable,
+    Mapping,
+    MutableMapping,
+    MutableSequence,
+    Optional,
+    Sequence,
+)
+
+from snuba.utils.streams.consumer import Consumer
+from snuba.utils.streams.types import Message, Partition, Topic, TPayload
+
+
+epoch = datetime(2019, 12, 19)
+
+
+class DummyConsumer(Consumer[TPayload]):
```

Comment:
Is this meant to be used only in tests ? If yes, why not moving it somewhere in the tests module ?

Reply:
Yeah, pretty much. My only hesitation with moving this from here is that I would consider this test infrastructure and not test code (in the sense that it is not a unit or integration test) and it should have a higher expectation of correctness — more similar to the code in the `snuba` directory than the code in the `tests` directory, basically. I think I have also had an assumption when we get to the point where we feel like we are able to turn type checking into a commit check, it would be applied to `snuba/` and not to `tests/`, and it's pretty important that this implementation does exactly mirror the actual Kafka backed implementation.

For example, one of the things that I would eventually like to do is actually parameterize a test against both this class and any actual consumer implementations (right now just Kafka, maybe eventually a file backend as we've discussed for performance testing the consumers) to ensure implementation parity. This is of course possible regardless of the import path of this class, but I think that kind of thinking makes me think it might belong here, rather than alongside the test that is verifying it's correctness?

I don't really have as strong of an opinion about this as the comment would probably imply — I'm fine moving it if you have a strong contrary opinion.

Reply:
ok, I like the argument around not type checking tests. OR at least turn type checking on for production before we fix all the tests.

Reply:
I guess another (admittedly more dubious) reason would be that if this ever becomes a module that is reused between this project and others, it'd be useful to include the testing backend with that move, and it'd need to be part of that package to be importable (not just from that project's tests) so this would slightly simplify that process and encourage it to not be forgotten during the refactor.